### PR TITLE
Clean up breadcrumb-item styles

### PR DESCRIPTION
**Pull request by mayank99**
- Made anchor links equal the same height as buttons.
- Removed need for explicitly specifying `iui-anchor` class. All `a` elements are now styled automatically.
- Applied margin on breadcrumb item instead of the element inside.
- Removed align-items: center as it was doing nothing.
- Made widths of all kinds of items consistent.

### CODE REVIEW

File: src/breadcrumbs/breadcrumbs.scss [hunk 1]
```diff
@@ -20,28 +20,27 @@
...
   }
 
-  > .iui-anchor,
-  > .iui-button > .iui-label {
+  > *,
+  .iui-label {
     overflow: hidden;
     white-space: nowrap;
     text-overflow: ellipsis;
-    max-width: $iui-3xl * 2;
```
Comment (bentleyvk) [inline src/breadcrumbs/breadcrumbs.scss:44]: Why removed?

File: src/breadcrumbs/breadcrumbs.scss [hunk 2]
```diff
@@ -20,28 +20,27 @@
...
+  height: $iui-component-height;
+  max-width: $iui-3xl * 2;
+  margin: 0 $iui-sm;
 
-  > .iui-anchor,
-  > span,
-  .iui-ellipsis {
-    margin: 0 $iui-sm;
+  > .iui-button {
+    margin: 0 (-$iui-s);
```
Comment (bentleyvk) [inline src/breadcrumbs/breadcrumbs.scss:29]: Why negative margin?

File: src/breadcrumbs/breadcrumbs.scss [hunk 3]
```diff
@@ -20,28 +20,27 @@
...
+  &:last-child {
+    &,
+    > .iui-button {
+      margin-right: 0;
+    }
   }
 
-  > .iui-anchor,
-  > .iui-button > .iui-label {
+  > *,
```
Comment (bentleyvk) [inline src/breadcrumbs/breadcrumbs.scss:39]: I don't like targeting everything. I would say to target `a` or bring back `iui-anchor`.

File: src/breadcrumbs/breadcrumbs.scss (hunk 1 above)
Comment (mayank99) [reply src/breadcrumbs/breadcrumbs.scss:44]: I placed it at the root of breadcrumb-item

File: src/breadcrumbs/breadcrumbs.scss (hunk 2 above)
Comment (mayank99) [reply src/breadcrumbs/breadcrumbs.scss:29]: Button has its own padding, so I'm counteracting the margin that I set above in the breadcrumb-item

File: src/breadcrumbs/breadcrumbs.scss (hunk 3 above)
Comment (mayank99) [reply src/breadcrumbs/breadcrumbs.scss:39]: This change is to allow targeting `a`, `span` and anything else like a custom `div`.

It's only first level `*`.

File: src/breadcrumbs/breadcrumbs.scss [hunk 4]
```diff
@@ -20,28 +20,27 @@
...
+  > .iui-button {
+    margin: 0 (-$iui-s);
   }
 
-  > .iui-button {
-    margin: 0 $iui-xs;
+  &:last-child {
+    &,
+    > .iui-button {
+      margin-right: 0;
```
Comment (veekeys) [inline src/breadcrumbs/breadcrumbs.scss:35]: Does it make any difference?

Comment (mayank99) [reply src/breadcrumbs/breadcrumbs.scss:35]: It increases the width, but looks same visually. I could remove

Comment (veekeys) [reply src/breadcrumbs/breadcrumbs.scss:35]: I dont see the width change with or without it

Comment (mayank99) [reply src/breadcrumbs/breadcrumbs.scss:35]: well yeah, i said "looks same visually" 😁

I've removed and pushed. See the change in test images

**Review by veekeys — APPROVED** [review]

**Pull request merged**